app/Http/Controllers/APIController.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Transaction;
+use Carbon\Carbon;
+
+class APIController extends Controller
+{
+    public function recentTransactions()
+    {
+        $todayMinusMonth = Carbon::now()->subMonth();
+
+        $recentTransactions = Transaction::where('date', '>=', $todayMinusMonth)
+            ->get();
+
+        return $recentTransactions;
+    }
+
+    private function createListingDate(Request $request)
+    {
+        if ($request->year != null) {
```

Comment:
is there any case in which the request D/M/Y would be null provided request isn't null? If not we should probably just check if the request itself is null?

Reply:
The only case to worry about if the supplied data is illegal, since this isn't customer facing shouldn't really have to worry about that as we'll be telling the user to select dates. If the data is illegal, it'll throw an exception

Reply:
I'll add a valdidation for the data in that case

Reply:
so in that case are the Y/M/D individual null checks even necessary? instead of a null request check?

Reply:
The null checks ensure that if the particular part of the request i.e. i don't provide a date, it defaults to day 1. Otherwise carbon will default to the current date as it tries to fill in the missing data.

The request itself will never be null, I'm only cofirming the parameters within the request itself.

Reply:
ahh right, didn't know that there was a possibility for those to be null!